pkg/ingester/ingester.go

```diff
@@ -661,19 +653,11 @@ func (i *Ingester) stopping(_ error) error {
 		}
 	}
 
-	if err := services.StopAndAwaitTerminated(context.Background(), i.compactionService); err != nil {
-		level.Warn(i.logger).Log("msg", "failed to stop TSDB Head compaction service", "err", err)
-	}
-
-	if err := services.StopAndAwaitTerminated(context.Background(), i.metricsUpdaterService); err != nil {
-		level.Warn(i.logger).Log("msg", "failed to stop metrics updater service", "err", err)
-	}
-
-	if err := services.StopAndAwaitTerminated(context.Background(), i.metadataPurgerService); err != nil {
-		level.Warn(i.logger).Log("msg", "failed to stop metadata purger service", "err", err)
+	if err := services.StopManagerAndAwaitStopped(context.Background(), i.subservicesBeforePartitionReader); err != nil {
+		level.Warn(i.logger).Log("msg", "failed to stop ingester subservices", "err", err)
 	}
 
-	if err := services.StopManagerAndAwaitStopped(context.Background(), i.subservices); err != nil {
+	if err := services.StopManagerAndAwaitStopped(context.Background(), i.subservicesAfterIngesterRingLifecycler); err != nil {
 		level.Warn(i.logger).Log("msg", "failed to stop ingester subservices", "err", err)
 	}
```

Comment:
We can make shutdown bit faster by stopping both managers at the same time:

```suggestion
	i.subservicesBeforePartitionReader.StopAsync()
	i.subservicesAfterIngesterRingLifecycler.StopAsync()

	if err := i.subservicesBeforePartitionReader.StartAsync(context.Background()); err != nil {
		level.Warn(i.logger).Log("msg", "failed to stop ingester subservices", "err", err)
	}
	if err := i.subservicesAfterIngesterRingLifecycler.StartAsync(context.Background()); err != nil {
		level.Warn(i.logger).Log("msg", "failed to stop ingester subservices", "err", err)
	}
```

Reply:
Is https://github.com/grafana/mimir/pull/7769/commits/170ef549ee5980531341b6d80e8468a810d16f3c what you meant?

Reply:
Yes, thanks